bin/network-monitor/src/deploy/mod.rs

```diff
@@ -0,0 +1,338 @@
+//! Account deployment module.
+//!
+//! This module contains functionality for deploying Miden accounts to the network.
+
+use std::collections::BTreeSet;
+use std::path::Path;
+use std::sync::Arc;
+use std::time::Duration;
+
+use anyhow::{Context, Result};
+use miden_lib::transaction::TransactionKernel;
+use miden_lib::utils::ScriptBuilder;
+use miden_node_proto::clients::{Builder, Rpc, RpcClient};
+use miden_node_proto::generated::shared::BlockHeaderByNumberRequest;
+use miden_node_proto::generated::transaction::ProvenTransaction;
+use miden_objects::account::{Account, AccountId, PartialAccount, StorageSlot};
+use miden_objects::assembly::{DefaultSourceManager, Library, LibraryPath, Module, ModuleKind};
+use miden_objects::asset::{AssetWitness, VaultKey};
+use miden_objects::block::{BlockHeader, BlockNumber};
+use miden_objects::crypto::merkle::{MmrPeaks, PartialMmr};
+use miden_objects::note::NoteScript;
+use miden_objects::transaction::{AccountInputs, InputNotes, PartialBlockchain, TransactionArgs};
+use miden_objects::{MastForest, Word};
+use miden_tx::auth::BasicAuthenticator;
+use miden_tx::utils::Serializable;
+use miden_tx::{
+    DataStore,
+    DataStoreError,
+    LocalTransactionProver,
+    MastForestStore,
+    TransactionExecutor,
+    TransactionMastStore,
+};
+use rand_chacha::ChaCha20Rng;
+use tokio::sync::Mutex;
+use tracing::instrument;
+use url::Url;
+
+use crate::COMPONENT;
+use crate::deploy::counter::{create_counter_account, save_counter_account};
+use crate::deploy::wallet::{create_wallet_account, save_wallet_account};
+
+pub mod counter;
+pub mod wallet;
+
+/// Ensure accounts exist, creating them if they don't.
+///
+/// This function checks if the wallet and counter account files exist.
+/// If they don't exist, it creates new accounts and saves them to the specified files.
+/// If they do exist, it does nothing.
+///
+/// # Arguments
+///
+/// * `wallet_file` - Path to the wallet account file.
+/// * `counter_file` - Path to the counter program account file.
+///
+/// # Returns
+///
+/// `Ok(())` if the accounts exist or were successfully created, or an error if creation fails.
+pub async fn ensure_accounts_exist(
+    wallet_filepath: &Path,
+    counter_filepath: &Path,
+    rpc_url: &Url,
+) -> Result<()> {
+    let wallet_exists = wallet_filepath.exists();
+    let counter_exists = counter_filepath.exists();
+
+    if wallet_exists && counter_exists {
+        tracing::info!("Account files already exist, skipping account creation");
+        return Ok(());
+    }
```

Comment:
I think this shouldn't matter too much outside of testing (where one may want to reset the state of at least one of the accounts), but if somehow only one of these were false, both accounts would be created again. Maybe this was intended but mentioning it just in case.